### PR TITLE
Use s3 for artifact storage

### PR DESCRIPTION
**Pull request by MancunianSam**

### CODE REVIEW

File: terraform/root.tf [hunk 1]
```diff
@@ -63,17 +63,17 @@ module "jenkins_backup_s3" {
   common_tags = local.common_tags
 }
 
-module "sonatype_intg" {
-  source      = "./modules/sonatype-build-task"
+module "s3_publish_intg" {
```
Comment (suzannehamilton) [inline terraform/root.tf:66]: Same question as here: https://github.com/nationalarchives/tdr-terraform-backend/pull/38/files#r403910570

File: docker/aws/Dockerfile [hunk 2]
```diff
@@ -5,4 +5,5 @@ RUN apt-get update -y && apt-get install -y python3 \
     && curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py \
     && python3 get-pip.py \
     && pip install boto3 \
-    && pip install awscli --upgrade --user
\ No newline at end of file
+    && pip install awscli --upgrade --user \
+    && pip install c7n c7n-mailer
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply docker/aws/Dockerfile:9]: Yeah, this was something for Paul, it doesn't really need a second pull request.

File: terraform/root.tf (hunk 1 above)
Comment (MancunianSam) [reply terraform/root.tf:66]: Yeah, fair point. I've changed it to just create the one.

File: docker/aws/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (suzannehamilton) [reply docker/aws/Dockerfile:9]: Ah, OK. 👍  It would have been helpful to put it in a separate commit to give it a bit more context.

**Pull request merged**